pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -403,15 +412,43 @@ public BrokerService(PulsarService pulsar, EventLoopGroup eventLoopGroup) throws
         this.defaultServerBootstrap = defaultServerBootstrap();
 
         this.pendingLookupRequests = ObserverGauge.build("pulsar_broker_lookup_pending_requests", "-")
-                .supplier(() -> pulsar.getConfig().getMaxConcurrentLookupRequest()
-                        - lookupRequestSemaphore.get().availablePermits())
+                .supplier(this::getPendingLookupRequest)
                 .register();
+        this.pendingLookupOperationsCounter = pulsar.getOpenTelemetry().getMeter()
+                .upDownCounterBuilder("pulsar.broker.topic.lookup.operation.pending.usage")
+                .setDescription("The number of pending lookup operations in the broker. "
+                        + "When it reaches threshold \"maxConcurrentLookupRequest\" defined in broker.conf, "
+                        + "new requests are rejected.")
+                .setUnit("{operation}")
+                .buildWithCallback(measurement -> measurement.record(getPendingLookupRequest()));
+        this.pendingLookupOperationsLimitGauge = pulsar.getOpenTelemetry().getMeter()
+                .gaugeBuilder("pulsar.broker.topic.lookup.operation.pending.limit")
```

Comment:
I need to figure out why on other examples in OTel they used UpDownCounter. For example: https://github.com/open-telemetry/semantic-conventions/blob/main/docs/runtime/jvm-metrics.md#metric-jvmmemorylimit

Reply:
Found this: https://opentelemetry.io/docs/specs/otel/metrics/supplementary-guidelines/

Reply:
I can presume that people may want a cluster-wide sum of topic lookup operations since they see that ZK is under stress and would like to know the cause, which a single broker does not cause. Once you define the pending usage as UpDownCounter, it only makes sense the limit is also UpDownCounter.